integration/simulation/validate_chain.go

```diff
@@ -377,35 +379,64 @@ func checkLogsReceived(t *testing.T, s *Simulation) {
 		sub.Unsubscribe()
 	}
 
-	for _, channel := range s.LogChannels {
-		checkReceivedHOCAndPOCLogs(t, channel)
+	logsReceived := 0
+	for owner, channel := range s.LogChannels {
+		logsReceived += checkReceivedHOCAndPOCLogs(t, owner, channel)
+	}
+	if logsReceived == 0 {
+		t.Errorf("no logs received during simulation")
 	}
 }
 
-// Checks that the owner has received the relevant HOC and POC logs.
-// TODO - #453 - Once the filtering by subscription ID is implemented, check client does not receive irrelevant logs.
-func checkReceivedHOCAndPOCLogs(t *testing.T, channel chan types.Log) {
-	var gotHOCEvent bool
-	var gotPOCEvent bool
+// Checks that the owner has only received relevant logs, and only from the HOC or POC contracts.
+func checkReceivedHOCAndPOCLogs(t *testing.T, owner string, channel chan types.Log) int {
+	logsReceived := 0
+
 	for {
 		select {
 		case receivedLog := <-channel:
-			if receivedLog.Address.Hex() == "0x"+bridge.HOCAddr {
-				gotHOCEvent = true
-			} else if receivedLog.Address.Hex() == "0x"+bridge.POCAddr {
-				gotPOCEvent = true
+			logsReceived++
+
+			if !isRelevant(owner, receivedLog) {
+				t.Errorf("received log that was not relevant (neither a lifecycle event not a relevant user event)")
 			}
-			if gotHOCEvent && gotPOCEvent {
-				// We have received both HOC and POC events. The test is successful.
-				return
+
+			logAddrHex := receivedLog.Address.Hex()
+			if logAddrHex != "0x"+bridge.HOCAddr && logAddrHex != "0x"+bridge.POCAddr {
+				t.Errorf("expected log from the HOC or POC contracts, but got a log from %s", logAddrHex)
 			}
 
 		// The logs will have built up on the channel throughout the simulation, so they should arrive immediately.
-		case <-time.After(time.Minute):
-			if !(gotHOCEvent && gotPOCEvent) {
-				t.Errorf("did not receive events for both the HOC and POC contracts")
-				return
-			}
+		default:
+			return logsReceived
+		}
+	}
+}
+
+// Checks whether the log is relevant to the recipient (either a lifecycle event or a relevant user event).
+func isRelevant(owner string, receivedLog types.Log) bool {
```

Comment:
We replicate a lot of the logic for deciding relevancy in the `SubscriptionManager` here.